tools/caas-tag-selector/tag-selector.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-underscore-dangle, import/no-unresolved */
+import 'https://da.live/nx/public/sl/components.js';
+import { LitElement, html } from 'https://da.live/nx/deps/lit/lit-core.min.js';
+import getStyle from 'https://da.live/nx/utils/styles.js';
+import DA_SDK from 'https://da.live/nx/utils/sdk.js';
+import {
+  getTags,
+  getAemRepo,
+  getRootTags,
+  jcrTitle,
+  caasContentType,
+  caasProducts,
+} from '../tags/tag-utils.js';
+
+const style = await getStyle(import.meta.url);
+const { context, token } = await DA_SDK.catch(() => null);
+const options = { headers: { Authorization: `Bearer ${token}` } };
+const collectionName = 'dx-tags/dx-caas';
+
+async function processRootTags(opts) {
+  const aemConfig = await getAemRepo(context, opts).catch(() => null);
+  const errorEvent = (message) => {
+    const error = new CustomEvent('caasOptionsError', { details: { message } });
+    document.dispatchEvent(error);
+  };
+
+  if (!aemConfig || !aemConfig.aemRepo) {
+    const repoError = 'Error in retrieving AemRepo';
+    errorEvent(repoError);
```

Comment:
Why not return an empty array at this point since you know you can't get the tags